cheroot/server.py

```diff
@@ -1144,6 +1149,9 @@ class HTTPServer(object):
     nodelay = True
     """If True (the default since 3.1), sets the TCP_NODELAY socket option."""
 
+    drop_underscore_headers = False  # issue #17
```

Comment:
In general, I try to avoid solutions that involve boolean flags, especially if those flags trickle through function call parameters.

What I'd rather see here is that instead of having a global `read_headers` function, that instead we consider creating a HeaderReader interface (or similar), and a given cheroot.server.Server can be configured with an instance of a HeaderReader (or some reference to one). In this way, cheroot provides an interface that allows this customization to be made or other customizations to be made without each one needing a boolean switch. I'll draft what I mean.